src/Sarif.WorkItems/SarifWorkItemsExtensions.cs

```diff
@@ -83,22 +83,54 @@ public static string CreateWorkItemTitle(this Run run)
                     (locationName == null ? "" : " (in " + locationName + ")");
         }
 
-        public static Dictionary<Run, int> ComputeRunResultCounts(this SarifLog log)
+        public static List<string> GetRunToolNames(this SarifLog log)
         {
-            if (log == null) { throw new ArgumentNullException(nameof(log)); }
-            if (log.Runs == null) { throw new ArgumentNullException(nameof(log.Runs)); }
+            return log?.Runs?.Where(r => r?.Results?.Count > 0).Select(r => r.Tool?.Driver?.Name).Distinct().ToList();
+        }
 
-            var resultCountsByRun = new Dictionary<Run, int>();
+        public static int GetAggregateResultCount(this SarifLog log)
+        {
+            return log?.Runs?.Select(r => r?.Results?.Count).Sum() ?? 0;
+        }
 
-            foreach (Run run in log?.Runs)
+        public static string PhraseToolNames(List<string> toolNames)
```

Comment:
>PhraseToolNames [](start = 29, length = 15)

This is a general purpose helper, not related to work item filing at all. So:

- Its name should be (for example) `MakePhraseWithAnd`. ("`MakeAndPhrase`" unfortunately sounds like it's doing two things, "making" and "phrasing".)
- Its parameter should be named (for example) `item` or `words`.
- Its local variable/return value should be named (for example) `phraseWithAnd`.
- It should be an extension method on `IEnumerable<string>`.
- It should live in src\Sarif\ExtensionMethods.cs.

Reply:
On second thought, how about `ToAndPhrase` as the name of the extension method. `words.ToAndPhrase()` seems like it reads nicely.

---
In reply to: [398969489](https://github.com/microsoft/sarif-sdk/pull/1831#discussion_r398969489) [](ancestors = 398969489)

Reply:
I was wondering where the appropriate place to put this was.  As you commented on elsewhere, the fact that the testcase was not naturally calling it as an extension method seemed to imply something was off.

I definitely don't like MakeAndPhrase simply because I did at a glance read it as two different things.  ToAndPhrase seems fine to me.

---
In reply to: [398994168](https://github.com/microsoft/sarif-sdk/pull/1831#discussion_r398994168) [](ancestors = 398994168,398969489)